### DISCUSSION TITLE
What's the difference between the `async_middleware` and the `m` macro

### DISCUSSION THREAD

**Discussion by heliomar-pena** [thruster-rs/Thruster#224] [Q&A] [answered]
I checked and exist two functions so similar: `async_middleware` and `m`

![image](https://user-images.githubusercontent.com/66505715/188292886-35b63cec-303f-4d26-8b94-88ccf47b558e.png)

Also, I noticed the [example in Readme.md](https://github.com/thruster-rs/Thruster#error-handling) is importing `async_middleware` but is using `m`, which is a little confusing to me. Is there any difference between them?

**Comment by trezm** [comment, marked as answer]
Good question, I should have marked `async_middleware` as deprecated. The way forward (including in examples) _should_ be just using the `m` middleware!!

**Comment by trezm** [comment]
it _should_ be updated in [this PR](https://github.com/thruster-rs/Thruster/pull/225). Feel free to take a look!

**Comment by heliomar-pena** [comment]
Thanks for clarify. I checked and your PR looks good
